created torch.reshape term entry (#5129)
Codecademy/docs: content/pytorch/concepts/tensors/reshape/reshape.md

---
Title: 'Pytorch Tensors Reshape'
Description: 'Torch.reshape returns a tensor with the same data and number of elements as input, but with a specified shape.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
  - 'Web Development'
Tags:
  - 'Modules'
  - 'Methods'
CatalogContent:
  - 'learn-advanced-python-3'
  - 'paths/computer-science'
---

> **`Torch.reshape`** method reshapes a specified input that contains the same data and number of elements. When possible, the returned tensor will be a view of input, nevertheless, this behavior is not a guarantee. This allows the reshaping of data without copying.

Author: Thank you. I was wondering if I should or not write it in lowercase because it is at the beginning of the sentence.

## Syntax

`torch.reshape(input, shape)`: Returns a tensor with the same data and number of elements as input, but with a specified shape.

```pseudo
torch.reshape(input, shape)
```

## Example

Using the special value `-1`:

```py

> size = torch.tensor([10, 11], [12, 13])
> torch.reshape(size, (-1))

Reviewer: ```suggestion
size = torch.tensor([[10, 11], [12, 13]])
reshaped_size=torch.reshape(size, (-1))
```
Corrected the code, and added a pair of square brackets to the torch.tensor because it always uses a list of lists for clarity and consistency.
Also, I added a variable named `reshaped_size` to store the reshaped tensor and to print it. Always keep those few things in mind.

Author: Thank you. Unfortunately, I could not test the code.

```

The example above returns the following output:

```shell
tensor([10, 11, 12, 13])
```

> The tensor `size` has a shape of `(2, 2)`, meaning it has 4 elements in total. By using `(-1)` it is asked to PyTorch to reshape the tensor into a 1 dimension tensor with 4 elements, and keeping all the elements of the original tensor. The `-1` tells PyTorch to calculate the appropriate size for that dimension, which in this case is 4.

Reviewer: ``` suggestion
The tensor `size` has a shape of `(2, 2)`, meaning it has 4 elements in total. By using `(-1)` it is asked to PyTorch to reshape the tensor into a 1-dimensional tensor with 4 elements, keeping all the elements of the original tensor. The `-1` tells PyTorch to calculate the appropriate size for that dimension, which in this case is 4.
```

---

> Title: 'Pytorch Tensors Reshape'

Reviewer 2: ```suggestion
Title: '.reshape()'
```
You only need to add the function name as the title.

> Description: 'torch.reshape` returns a tensor with the same data and number of elements as the input tensor, but with a specified shape.'

Reviewer 2: ```suggestion
Description: 'Returns a tensor with the same data and number of elements as the input tensor, but with a specified shape.'` 
```

Subjects:
  - 'Computer Science'
  - 'Data Science'
  - 'Web Development'
Tags:
  - 'Modules'
  - 'Methods'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

> **`torch.reshape`** method reshapes a specified input tensor to a given shape while keeping the same data and number of elements. When possible, the returned tensor will have no data copied, however, this behavior is not guaranteed in all cases.

Reviewer 2: ```suggestion
The **`torch.reshape()`** method reshapes a specified input tensor to a given shape while keeping the same data and number of elements. When possible, the returned tensor will have no data copied, however, this behaviour is not guaranteed in all cases.
```

## Syntax

> `torch.reshape(input, shape)`: Returns a tensor with the same data and number of elements as input, but with a specified shape.

Reviewer 2: ```suggestion
```

```pseudo
torch.reshape(input, shape)
```

> 

Reviewer 2: Add the description for the parameters here, in this format:
```
- `parameter`: description
```

## Example

Using the special value `-1`:

```py

> size = torch.tensor([[10, 11], [12, 13]])
> reshaped_size = torch.reshape(size, (-1))

Reviewer 2: ```suggestion
import torch

# Define the tensor
size = torch.tensor([[10, 11], [12, 13]])

# Reshape the tensor
reshaped_size = torch.reshape(size, (-1,))  # Note the comma to make it a tuple

# Print the reshaped tensor
print(reshaped_size)
```
You should add the whole code. The print statement and a comma were missing while reshaping the tensor. I fixed both the issues.

```

The example above returns the following output:

```shell
tensor([10, 11, 12, 13])
```

> The tensor `size` has a shape of `(2, 2)`, meaning it has 4 elements in total. By using `(-1)` it is asked to PyTorch to reshape the tensor into a 1 dimensional tensor with 4 elements, keeping all the elements of the original tensor. The `-1` tells PyTorch to calculate the appropriate size for that dimension, which in this case is 4.

Reviewer 2: Move this line before the code.

## Codebyte Example

Run the following codes to understand how the `torch.reshape(input, shape)` method works:

```codebyte/python

> dimension = torch.tensor(10.)
> print(dimension)
> reshaped_dimension = torch.reshape(dimension, (2, 2))
> print(reshaped_dimension)

Reviewer: ```suggestion
dimension = torch.tensor(10.)
print(dimension)
reshaped_dimension = torch.reshape(dimension, (2, 2))
print(reshaped_dimension)
```

This code will cause an error as you attempt to reshape a tensor with a single element into a shape (2, 2).  this will raise an error because the number of elements in the original tensor does not match the total number of elements in the target shape. The total number of elements in the reshaped tensor must match the total number of elements in the original given tensor.

Author: @SaviDahegaonkar do you want me to replace the example or just delete this one and keep the second one?

> dimension = torch.tensor(10., 20., 30., 40.)
> print(dimension)
> reshaped_dimension = torch.reshape(dimension, (2, 2))
> print(reshaped_dimension)

Reviewer: ```suggestion
dimension = torch.tensor([10., 20., 30., 40.])
print(dimension)
reshaped_dimension = torch.reshape(dimension, (2, 2))
print(reshaped_dimension)
```
The `torch.tensor` function should be called with a list or tuple of values to create a tensor. Added `[ ]` to create a 1D tensor.

Author: Of course... I know it and completely missed it. Give me one second. Thank you

> dimension = torch.tensor([10., 20., 30., 40.])
> print(dimension)
> reshaped_dimension = torch.reshape(dimension, (2, 2))
> print(reshaped_dimension)

Reviewer 2: ```suggestion
import torch

dimension = torch.tensor([10., 20., 30., 40.])
print(dimension)
reshaped_dimension = torch.reshape(dimension, (2, 2))
print(reshaped_dimension)
```

```

```codebyte/python

> main_tensor = torch.tensor([[6, 6, 6], [6, 6, 6]])
> print(main_tensor)
> reshaped_tensor = torch.reshape(main_tensor, (3,2))
> print(reshaped_tensor)

Reviewer: ```suggestion
main_tensor = torch.tensor([[6, 6, 6], [6, 6, 6]])
print(main_tensor)
reshaped_tensor = torch.reshape(main_tensor, (3,2))
print(reshaped_tensor)

```
This code is absolutely correct .

> main_tensor = torch.tensor([[6, 6, 6], [6, 6, 6]])
> print(main_tensor)
> reshaped_tensor = torch.reshape(main_tensor, (3,2))
> print(reshaped_tensor)

Reviewer 2: ```suggestion
import torch

main_tensor = torch.tensor([[6, 6, 6], [6, 6, 6]])
print(main_tensor)
reshaped_tensor = torch.reshape(main_tensor, (3,2))
print(reshaped_tensor)
```
The whole code should run as it is.
